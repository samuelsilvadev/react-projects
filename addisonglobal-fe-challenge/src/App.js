import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Home from './Home';

import Header from './components/header/Header';

import './App.scss';

const App = () => (
	<>
		<Header />
		<Switch>
			<Route exact path="/" component={Home} />
		</Switch>
	</>
);

export default App;
