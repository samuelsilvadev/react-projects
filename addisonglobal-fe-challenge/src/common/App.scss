* {
	box-sizing: border-box;
	margin: 0;
	padding: 0;
}

body {
	margin: 0;
	padding: 0;
	font-family: sans-serif;
}

.main {
	padding: 0 15px;
}
