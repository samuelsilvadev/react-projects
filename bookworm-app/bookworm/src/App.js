import React from 'react';
import { Route } from 'react-router-dom';
import PropTypes from 'prop-types';
import HomePage from './components/pages/HomePage'
import LoginPage from './components/pages/LoginPage';
import DashBoardPage from './components/pages/DashBoardPage';
import UserRoute from './components/routes/UserRoute';
import GuestRoute from './components/routes/GuestRoute';
import SignUp from './components/pages/SignUpPage';

const App = ({ location }) => (
  <div className="ui container">
    <Route location={location} path="/" exact component={HomePage} />
    <GuestRoute location={location} path="/login" component={LoginPage} />
    <GuestRoute location={location} path="/signup" component={SignUp} />
    <UserRoute location={location} path="/dashboard" component={DashBoardPage} />
  </div>
);

App.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired
};

export default App;
