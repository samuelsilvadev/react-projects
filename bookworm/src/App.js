import React from 'react';

const App = () => (
  <div>Hi</div>
);

export default App;
