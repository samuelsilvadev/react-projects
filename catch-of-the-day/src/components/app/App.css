.main {
	display: flex;
	height: 90vh;
	margin: 0 auto;
	margin-top: 5vh;
}

.header,
.order,
.inventory {
	flex: 1 1 33.33%;
	padding: 2rem;
	border: 1rem double #1a1a1a;
	position: relative;
	background: #fff;
	box-shadow: 0px 5px 5px rgba(0, 0, 0, 0.1);
	overflow: scroll;
}
