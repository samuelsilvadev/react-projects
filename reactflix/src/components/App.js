import React from 'react';

import Title from './App.style';

const App = () => (
	<main>
		<Title data-enzyme-id="app-title">React Flix</Title>
	</main>
);

export default App;
