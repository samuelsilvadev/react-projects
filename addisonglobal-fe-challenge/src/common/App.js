import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { Home } from './pages';

import Header from './components/header/Header';
import BetSlip from './components/betslip';

import { useAsideContext, withAsideContext } from './contexts/AsideContext';

import './App.scss';

const App = () => {
	const { handleOpen, handleClose, isAsideVisible } = useAsideContext();

	return (
		<>
			<Header onAsideOpen={handleOpen} />
			<main className="main">
				<Switch>
					<Route exact path="/" component={Home} />
				</Switch>
				<BetSlip open={isAsideVisible} onClose={handleClose} />
			</main>
		</>
	);
};

export default withAsideContext(App);
