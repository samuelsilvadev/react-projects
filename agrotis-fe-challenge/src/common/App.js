import React from 'react';
import styled from 'styled-components';
import { Route, Switch } from 'react-router-dom';

import { Header, HEADER_HEIGHT } from './components';
import Home from './pages/Home';
import { GlobalStyle } from './Global.style';
import { gteSmallMedia } from './media.style';

const Main = styled.main`
	background-color: #ecf0f1;
	min-height: calc(100vh - ${HEADER_HEIGHT});
	padding: 2rem;

	${gteSmallMedia} {
		padding: 2rem 4rem;
	}
`;

const App = () => (
	<>
		<GlobalStyle />
		<Header />
		<Main>
			<Switch>
				<Route exact path="/" component={Home} />
			</Switch>
		</Main>
	</>
);

export default App;
