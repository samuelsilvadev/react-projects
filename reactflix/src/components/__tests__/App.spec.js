import React from 'react';
import { shallow } from 'enzyme';

import App from '../App';

describe('<App />', () => {
	it('should render correctly with title', () => {
		const tree = shallow(<App />);
		const $title = tree.find('[data-enzyme-id="app-title"]');

		expect($title).toBeDefined();
		expect($title.text()).toBe('React Flix');
		expect(tree).toMatchSnapshot();
	});
});
