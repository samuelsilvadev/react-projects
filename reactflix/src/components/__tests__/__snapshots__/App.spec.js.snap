// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly with title 1`] = `ShallowWrapper {}`;
